Mathlib/Combinatorics/SimpleGraph/Acyclic.lean

```diff
@@ -197,4 +198,89 @@ lemma IsTree.card_edgeFinset [Fintype V] [Fintype G.edgeSet] (hG : G.IsTree) :
       refine (hG.existsUnique_path _ _).unique ((hf _).takeUntil _) ?_
       simp [h.ne]
 
+theorem maximal_isAcyclic_iff_isTree [Nonempty V] :  Maximal IsAcyclic G ↔ G.IsTree where
+  mp := by
```

Comment:
This may be a matter of taste, but: I'd write this iff proof using tactic mode (and e.g. `constructor` as the first tactic), instead of using `where` and `mp`, `mpr`.